crates/algokit_utils/src/clients/asset_manager.rs

```diff
@@ -0,0 +1,162 @@
+use algod_client::apis::{AlgodClient, Error as AlgodError};
+use algokit_transact::Address;
+use std::sync::Arc;
+
+#[derive(Debug, Clone)]
+pub struct BulkAssetOptInOutResult {
+    pub asset_id: u64,
+    pub transaction_id: String,
+}
+
+#[derive(Debug, Clone)]
+pub struct AccountAssetInformation {
+    pub asset_id: u64,
+    pub balance: u64,
+    pub frozen: bool,
+    pub round: u64,
+}
+
+#[derive(Debug, Clone)]
+pub struct AssetInformation {
```

Comment:
Should we keep dedicated assetinfo class to just return raw type from algod? in ts and py it was relevant because we wanted to abstract away algosdk dependencies but in our case we have new modular building blocks - kept asset info explicit here for now but can revert to returning raw algod client type

Reply:
tweaked the api generator to have name mappings that we can control from the js pre processing script - custom asset info type removed

Reply:
Nice!